R/Data-methods.R

```diff
@@ -231,84 +231,75 @@ setMethod(
-              object@x <- c(object@x,
-                            rep(x,
-                                length(y)))
-
-              ## add DLT data
-              object@y <- c(object@y, as.integer(y))
-
-              ## add ID
-              object@ID <- c(object@ID, ID)
-
-              ## add cohort number
-              if(newCohort){
-                  object@cohort <- c(object@cohort,
-                                     rep(max(tail(object@cohort, 1L), 0L) + 1L,
-                                         length(y)))
-              }else{
-                  object@cohort <- c(object@cohort,
-                                     rep(max(tail(object@cohort, 1L), 0L),
-                                         length(y)))
-              }
+# Data-update ----
 
-              ## return the object
-              return(object)
-          })
+#' Update method for the [`Data`] class.
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' A method that updates existing [`Data`] object with new data.
+#'
+#' @param object (`Data`)\cr object you want to update.
+#' @param x (`numeric`)\cr the dose level (one level only!).
+#' @param y (`integer`)\cr the DLT vector (0/1 vector) for all patients in this cohort.
+#'   You can also supply `numeric` vectors, but these will then be converted to
+#'   `integer` internally.
+#' @param ID (`integer`)\cr the patient IDs.
+#'   You can also supply `numeric` vectors, but these will then be converted to
+#'   `integer` internally.
+#' @param new_cohort (`flag`)\cr if `TRUE` (default) the new data are assigned to a new cohort.
+#' @param \dots not used.
+#'
+#' @return The new, updated [`Date`] object.
+#'
+#' @aliases update-Data-method
+#' @example examples/Data-method-update-Data.R
+#' @export
+#'
+setMethod(
+  f = "update",
+  signature = signature(object = "Data"),
+  definition = function(object,
+                        x,
+                        y,
+                        ID = length(object@ID) + seq_along(y),
+                        new_cohort = TRUE,
+                        ...) {
+    y_len <- length(y)
+
+    assert_number(x)
+    assert_numeric(y, lower = 0, upper = 1, min.len = 1)
+    assert_numeric(ID, len = y_len)
+    assert_disjunct(object@ID, ID)
+    assert_flag(new_cohort)
+
+    # Which grid level is the dose?
+    gridLevel <- matchTolerance(x, object@doseGrid)
+    object@xLevel <- c(object@xLevel, rep(gridLevel, y_len))
+
+    # Increment sample size.
+    object@nObs <- object@nObs + y_len
+
+    # Add dose.
+    object@x <- c(object@x, rep(as.numeric(x), y_len))
+
+    # Add DLT data.
+    object@y <- c(object@y, safeInteger(y))
+
+    # Add ID.
+    object@ID <- c(object@ID, safeInteger(ID))
+
+    # Add cohort number.
+    last_cohort <- max(tail(object@cohort, 1L), 0L)
+    object@cohort <- c(
+      object@cohort,
+      rep(last_cohort, y_len) + ifelse(new_cohort, 1L, 0L)
+    )
 
+    stopifnot(test_class(object, "Data"), validObject(object))
```

Comment:
```suggestion
    validObject(object)
```
should be sufficient as we know the class and this will raise an error if not